drivers/mspi/mspi_dw_vendor_specific.h

```diff
@@ -97,7 +101,217 @@ static inline int vendor_specific_xip_disable(const struct device *dev,
+						 config->vendor_specific_data;
+
+
+	QSPI_TRANSFER_LIST_Type *transfer_list = vendor_data->transfer_list;
+	EVDMA_JOB_Type *joblist = vendor_data->joblist;
+
+	int tmod = 0;
+	int job_idx = 0;
+
+	if (packet->dir == MSPI_TX) {
+		preg->CONFIG.RXTRANSFERLENGTH = 0;
+
+		/* Setting up EVDMA joblist depending on cmd, addr and data */
+
+		/*
+		 * Command address will always have a length of 4 from the DMA's perspective,
+		 * QSPI peripheral will use length of data specified in core registers
+		 */
+		if (dev_data->xfer.cmd_length > 0) {
+			joblist[job_idx++] = EVDMA_JOB(&packet->cmd, 4, EVDMA_ATTRIBUTE);
+		}
+		if (dev_data->xfer.addr_length > 0) {
+			joblist[job_idx++] = EVDMA_JOB(&packet->address, 4, EVDMA_ATTRIBUTE);
+		}
+		if (packet->num_bytes > 0) {
+			joblist[job_idx++] = EVDMA_JOB(packet->data_buf, packet->num_bytes,
+						       EVDMA_ATTRIBUTE);
+		}
+		/* Always terminate with null job */
+		joblist[job_idx] = EVDMA_NULL_JOB();
+		/* tx_job should point to first valid job, or null if none */
+		if (job_idx > 0) {
+			transfer_list->tx_job = &joblist[0];
+		} else {
+			transfer_list->tx_job = &joblist[job_idx];
+		}
+
+		/* rx_job always EVDMA_NULL_JOB() for transmit */
+		transfer_list->rx_job = &joblist[job_idx];
+		tmod = 0;
+	} else {
+		preg->CONFIG.RXTRANSFERLENGTH = ((packet->num_bytes + dev_data->xfer.addr_length +
+						dev_data->xfer.cmd_length) >>
+						dev_data->bytes_per_frame_exp) - 1;
+		joblist[0] = EVDMA_JOB(packet->data_buf, packet->num_bytes, EVDMA_ATTRIBUTE);
+		joblist[1] = EVDMA_NULL_JOB();
+		transfer_list->tx_job = &joblist[1];
+		transfer_list->rx_job = &joblist[0];
+
+		tmod = QSPI_TMOD_RX_ONLY;
+	}
+
+	/*
+	 * In slave mode, a tmod register in the wrapper also needs to be set. Currently
+	 * the address not in MDK so temp fix.
+	 */
+	uintptr_t tmod_addr = (uintptr_t)preg + QSPI_TMOD_OFFSET;
+
+	sys_write32(tmod, tmod_addr);
+
+	preg->CONFIG.TXBURSTLENGTH = (config->tx_fifo_depth_minus_1+1)-config->dma_tx_data_level;
+	preg->CONFIG.RXBURSTLENGTH = config->dma_rx_data_level+1;
+	preg->DMA.CONFIG.LISTPTR = (uint32_t)transfer_list;
+
+	preg->INTENSET = BIT(QSPI_INTENSET_CORE_Pos)
+		       | BIT(QSPI_INTENSET_DMADONE_Pos);
+
+	preg->TASKS_START = 1;
+}
+
+static inline bool vendor_specific_dma_accessible_check(const struct device *dev,
+							const uint8_t *data_buf)
+{
+	const struct mspi_dw_config *config = dev->config;
+	NRF_QSPI_Type *preg = (NRF_QSPI_Type *)config->wrapper_regs;
+
+	return nrf_dma_accessible_check(preg, data_buf);
+}
+
+static inline bool vendor_specific_read_dma_irq(const struct device *dev)
+{
+	const struct mspi_dw_config *config = dev->config;
+	NRF_QSPI_Type *preg = (NRF_QSPI_Type *)config->wrapper_regs;
+
+	return (bool) preg->EVENTS_DMA.DONE;
+}
+
+#else /* Supply empty vendor specific macros for generic case */
+
+/* Empty macros for generic case - no vendor-specific data */
+#define VENDOR_SPECIFIC_DATA_DEFINE(inst)
+#define VENDOR_SPECIFIC_DATA_GET(inst) NULL
```

Comment:
Instead of repeating the empty definitions for every case where vendor specific data is not used, we could have at the end of this file:
```
#ifndef VENDOR_SPECIFIC_DATA_DEFINE
#define VENDOR_SPECIFIC_DATA_DEFINE(inst)
...
```